src/System.Memory/src/System/SpanExtensions.cs

```diff
@@ -295,5 +295,47 @@ public static void CopyTo<T>(this T[] array, Span<T> destination)
         {
             new ReadOnlySpan<T>(array).CopyTo(destination);
         }
+
+        /// <summary>
+        /// Determines whether two sequences overlap.
+        /// </summary>
+        public static unsafe bool Overlaps<T>(this Span<T> first, ReadOnlySpan<T> second)
```

Comment:
It doesn't implicitly convert a Span to ReadOnlySpan when using extension method syntax. But I leave it up to the experts to decide which overloads are really needed.